emacs-pdf-tools-0.80 fails to build

## Issue description

I'm using `(emacsPackagesNgGen emacs).emacsWithPackages (epkgs: [epkgs.pdf-tools])`. This fails to build currently because:

```
poppler-hack.cc: In function 'void xpoppler_annot_set_rectangle(PopplerAnnot*, PopplerRectangle*)':
poppler-hack.cc:88:49: error: invalid conversion from 'const GooString*' to 'GooString*' [-fpermissive]
     GooString *state = a->annot->getAppearState ();
                        ~~~~~~~~~~~~~~~~~~~~~~~~~^~
poppler-hack.cc: In function 'gchar* xpoppler_annot_markup_get_created(PopplerAnnotMarkup*)':
poppler-hack.cc:108:27: error: invalid conversion from 'const GooString*' to 'GooString*' [-fpermissive]
     text = annot->getDate ();
            ~~~~~~~~~~~~~~~^~
make[2]: *** [Makefile:556: epdfinfo-poppler-hack.o] Error 1
make[2]: Leaving directory '/build/source/server'
```

## Technical details

 - system: `"x86_64-linux"`
 - host os: `Linux 4.14.43, NixOS, 18.09pre140958.696c6bed4e8 (Jellyfish)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.0.2`
 - channels(philipp): `"nixpkgs-18.09pre140731.c29d2fde74d"`
 - channels(root): `"nixos"`
 - nixpkgs: `/home/philipp/.nix-defexpr/channels/nixpkgs`

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -15702,7 +15702,7 @@ with pkgs;
 
   emacsPackagesNgGen = emacs: import ./emacs-packages.nix {
     inherit lib newScope stdenv;
-    inherit fetchFromGitHub fetchgit fetchhg fetchurl;
+    inherit fetchFromGitHub fetchgit fetchhg fetchurl fetchpatch;
     inherit emacs texinfo makeWrapper runCommand;
     inherit (xorg) lndir;
 

--- a/pkgs/top-level/emacs-packages.nix
+++ b/pkgs/top-level/emacs-packages.nix
@@ -32,7 +32,7 @@
 #   `meta` with `platforms` and `homepage` set to something you are
 #   unlikely to want to override for most packages
 
-{ lib, newScope, stdenv, fetchurl, fetchgit, fetchFromGitHub, fetchhg, runCommand
+{ lib, newScope, stdenv, fetchurl, fetchgit, fetchFromGitHub, fetchhg, fetchpatch, runCommand
 
 , emacs, texinfo, lndir, makeWrapper
 , trivialBuild
@@ -82,6 +82,20 @@ let
       rev = "v${version}";
       sha256 = "1i4647vax5na73basc5dz4lh9kprir00fh8ps4i0l1y3ippnjs2s";
     };
+    patches = [
+      (fetchpatch {
+        url = https://github.com/politza/pdf-tools/commit/6505a0e817495b85897c9380161034ae611ddd90.patch;
+        sha256 = "122ycbja8ckaysp58xqfcv11sgpbcp78pll5mywf9hgr0qap9jsy";
+      })
+      (fetchpatch {
+        url = https://github.com/politza/pdf-tools/commit/ded6341b0e3ad97e8b14f68c1796ba66dc155fd1.patch;
+        sha256 = "0hd2v4c6xq2jzg2m6s5kzs0fldgygf1pnfqd11v6x4w05zvxn6a2";
+      })
+      (fetchpatch {
+        url = https://github.com/politza/pdf-tools/commit/50a5297b82e26cfd52f6c00645ddc1057099d6a7.patch;
+        sha256 = "107rqzldg06h8k3pmdinkl78dr4xycm570sp2an4ihjmpmph0z39";
+      })
+    ];
     nativeBuildInputs = [ external.pkgconfig ];
     buildInputs = with external; [ autoconf automake libpng zlib poppler ];
     preBuild = "make server/epdfinfo";

